jupyter_client/multikernelmanager.py

```diff
@@ -104,10 +109,28 @@ def _find_available_port(self, ip):
 
                 return port
 
+    shared_context = Bool(
+        True,
+        config=True,
+        help="Share a single zmq.Context to talk to all my kernels",
+    )
+
+    _created_context = Bool(False)
+
     context = Instance('zmq.Context')
+
+    @default("context")
     def _context_default(self):
+        self._created_context = True
         return zmq.Context()
 
+    def __del__(self):
+        if self._created_context and self.context and not self.context.closed:
+            if self.log:
+                self.log.debug("Destroying zmq context for %s", self)
+            self.context.destroy()
+        super().__del__()
```

Comment:
Oops. Need to handle super not having defined del (I thought this should have been on all objects...):


```suggestion
        try:
            super_del = super().__del__
        except AttributeError:
            pass
        else:
            super_del()
```

Reply:
Done